src/state-model.js

```diff
@@ -16,27 +18,27 @@ var StateMixin = {
      * {@link StateMixin.preinitialize}, call this method once in your own
      * constructor, `preinitialize` or `initialize` method. */
     bindStateEvents: function() {
-        this.on('change', this.broadcastStateEvents);
+        this.on('all', this.broadcastStateEvents);
```

Comment:
(That's what our cat Suki has to say about this line)

Reply:
For my own understanding: does taking `all` StateEvents and then filtering for `change` using a regex yield other results than using `.on('change')`

(I would assume so, but it is unclear to me what the difference in scope is, exactly.)

Reply:
Ah, very astute question! Yes, it is subtle. `change:xxx` is distinct from `change`.

Suppose that I have a model with content `{awake: false}`. If I `.set({awake: true, hungry: true})` that model, three events will be triggered:

- `change:awake`
- `change:hungry`
- `change`

Hence, listening for `all` and then filtering for `change:` causes the listener to run for every individual attribute change, instead of running once at the end of a batch of changes. This ia also the reason why a regex is needed: it cannot know all possible attribute names in advance.

The way in which this avoids infinite loops is difficult to wrap your head around. It is related to the fact that handlers of these events can cause nested calls to `set`, i.e. changes during changes. Maybe it will be clearer now, if you reread the test.